asv_bench/benchmarks/benchmarks.py

```diff
@@ -67,140 +67,134 @@ def execute(df):
 
 
 class BaseTimeGroupBy:
-    def setup(self, data_size, ncols=1):
-        self.df = generate_dataframe(
-            ASV_USE_IMPL, "int", data_size[1], data_size[0], RAND_LOW, RAND_HIGH
+    def setup(self, shape, groupby_ncols=1):
+        self.df, self.groupby_columns = generate_dataframe(
+            ASV_USE_IMPL,
+            "int",
+            *shape,
+            RAND_LOW,
+            RAND_HIGH,
+            groupby_ncols,
+            count_groups=100 if ASV_DATASET_SIZE == "big" else 5,
```

Comment:
what do you think about replacing all this `if ASV_DATASET_SIZE == smt` to the dictionary of shapes/sizes/group_counts etc...
```python
UNARY_OP_DATA_SIZE = {
    "big":  big_shapes,
    "normal": normal_shapes,
}
...
class BaseTimeGroupBy:
    count_groups = {
        "big": 100,
        "normal": 5,
    }
...
```

Reply:
Updated on your approach.